modules/lvgl/lvgl.c

```diff
@@ -221,50 +230,53 @@ lv_result_t lv_mem_test_core(void)
+	disp_data_##n.display_dev = display_dev_##n;                                               \
+	display_get_capabilities(display_dev_##n, &disp_data_##n.cap);                             \
+	lvgl_display_##n =                                                                         \
+		lv_display_create(disp_data_##n.cap.x_resolution, disp_data_##n.cap.y_resolution); \
+	if (lvgl_display_##n == NULL) {                                                            \
+		printk("Failed to create LV display object.\n");                                   \
+		return -ENOMEM;                                                                    \
+	}                                                                                          \
+	lv_display_set_user_data(lvgl_display_##n, &disp_data_##n);                                \
+	if (set_lvgl_rendering_cb(lvgl_display_##n) != 0) {                                        \
+		printk("Display is not supported.\n");                                             \
+		return -ENOTSUP;                                                                   \
+	}                                                                                          \
+                                                                                                   \
+	IF_ENABLED(CONFIG_LV_Z_BUFFER_ALLOC_STATIC,						   \
+			(LVGL_ALLOCATE_RENDERING_BUFFERS(lvgl_display_##n, n)))                    \
+	IF_DISABLED(CONFIG_LV_Z_BUFFER_ALLOC_STATIC,						   \
+		(err = lvgl_allocate_rendering_buffers(lvgl_display_##n);			   \
+		if (err < 0) { return err;			                                   \
+		}))                 \
+	IF_ENABLED(CONFIG_LV_Z_FULL_REFRESH,							   \
+		(lv_display_set_render_mode(lvgl_display_##n, LV_DISPLAY_RENDER_MODE_FULL);))
+
 int lvgl_init(void)
 {
-	const struct device *display_dev = DEVICE_DT_GET(DISPLAY_NODE);
-
 	int err = 0;
 
-	if (!device_is_ready(display_dev)) {
-		LOG_ERR("Display device not ready.");
-		return -ENODEV;
-	}
-
 #if CONFIG_LV_Z_LOG_LEVEL != 0
 	lv_log_register_print_cb(lvgl_log);
 #endif
 
+	LOG_DBG("LVGL initialize");
 	lv_init();
 	lv_tick_set_cb(k_uptime_get_32);
 
 #ifdef CONFIG_LV_Z_USE_FILESYSTEM
 	lvgl_fs_init();
-#endif
+#endif /* CONFIG_LV_Z_USE_FILESYSTEM */
 
-	disp_data.display_dev = display_dev;
-	display_get_capabilities(display_dev, &disp_data.cap);
-
-	display = lv_display_create(disp_data.cap.x_resolution, disp_data.cap.y_resolution);
-	if (!display) {
-		return -ENOMEM;
-	}
-	lv_display_set_user_data(display, &disp_data);
-
-	if (set_lvgl_rendering_cb(display) != 0) {
-		LOG_ERR("Display not supported.");
-		return -ENOTSUP;
-	}
-
-	err = lvgl_allocate_rendering_buffers(display);
-	if (err != 0) {
-		return err;
-	}
-
-#ifdef CONFIG_LV_Z_FULL_REFRESH
-	lv_display_set_render_mode(display, LV_DISPLAY_RENDER_MODE_FULL);
-#endif
+	FOR_EACH(LV_DISPLAY_INIT, (), LV_DISPLAYS_TO_INITIALIZE)
+		;
```

Comment:
Not needed space.

Reply:
Hmm that is strange :D, I'd prefer living with the complaint and do a override on the merge.

Reply:
just add a pair of `/* clang-format off */` `/* clang-format on */` marker, clang-format makes a mess with the macro quite often, no shame in formatting blocks manually